internal/pool/conn_check.go

```diff
@@ -0,0 +1,49 @@
+// +build linux darwin dragonfly freebsd netbsd openbsd solaris illumos
+
+package pool
+
+import (
+	"errors"
+	"io"
+	"net"
+	"syscall"
+	"time"
+)
+
+var errUnexpectedRead = errors.New("unexpected read from socket")
+
+func connCheck(conn net.Conn) error {
+	//reset deadline
+	_ = conn.SetDeadline(time.Time{})
```

Comment:
@monkey92t I believe this line resolves https://github.com/go-redis/redis/pull/1849/files. I've been using similar code with go-clickhouse and everything seems to be working fine.